src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -2027,6 +2326,30 @@ public partial interface IResourceCollection : System.Collections.Generic.IList<
     {
     }
 
+    public partial interface IResourceExecutionConfigurationBuilder
+    {
+        IResourceExecutionConfigurationBuilder AddExecutionConfigurationGatherer(IResourceExecutionConfigurationGatherer gatherer);
+        System.Threading.Tasks.Task<(IProcessedResourceExecutionConfiguration, System.Exception?)> BuildAsync(DistributedApplicationExecutionContext executionContext, Microsoft.Extensions.Logging.ILogger? resourceLogger = null, System.Threading.CancellationToken cancellationToken = default);
+    }
+
+    public partial interface IResourceExecutionConfigurationData
+    {
+    }
+
+    public partial interface IResourceExecutionConfigurationGatherer
```

Comment:
The naming here gives me Java feels.

Reply:
@davidfowl @danegsta @karolz-ms - what do you think about dropping `Resource` and all these types become:

* `IExecutionConfiguration*`

Reply:
I like that